### PR TITLE
urbanvanilla keymap for drop/alt

### PR DESCRIPTION
**Pull request by urbanvanilla**
## Description

Keymap with simple changes all in userspace for the drop/alt.
Modified matrix.c in the wait_us function to get rid of annoying key chatter that plagues this keyboard.
Implemented a really simple RGB timer that turns off RGB Matrix lighting after a set time.
Changed escape key to grave escape.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/massdrop/alt/keymaps/urbanvanilla/rules.mk [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+VIRTSER_ENABLE = no         # USB Serial Driver
+RAW_ENABLE = no             # Raw device
+AUTO_SHIFT_ENABLE = no      # Auto Shift
+
+# Custom RGB matrix handling
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = custom
+WPM_ENABLE = yes
+VELOCIKEY_ENABLE = yes # doesn't work with RGB MATRIX only RGB LIGHT :(
```
Comment (drashna) [inline keyboards/massdrop/alt/keymaps/urbanvanilla/rules.mk:38]: Everything but these two lines should be removed.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/massdrop/alt/keymaps/urbanvanilla/matrix.c [hunk 2]
```diff
@@ -0,0 +1,176 @@
+/*
```
[the remark this answers is not in the document]
Comment (urbanvanilla) [reply keyboards/massdrop/alt/keymaps/urbanvanilla/matrix.c:1]: this file has the wait_us() function modification in the uint8_t matrix_scan() function that actually makes the change for key chatter with this keyboard. I've increased it in my latest update. But would it be customary to just put the single function and remove all other code?

**Review by urbanvanilla — COMMENTED** [review]
made the changes as reviewed, except for the matrix.c not sure if I should clear it all and just have the one function or keep it all with that minor change.

File: keyboards/massdrop/alt/keymaps/urbanvanilla/matrix.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/massdrop/alt/keymaps/urbanvanilla/matrix.c:1]: You mean this delay? 
https://github.com/qmk/qmk_firmware/blob/985b8cab7ca7103aee2b83696824683bd5159888/quantum/matrix.c#L152

If so, that can be set by defining `MATRIX_IO_DELAY`, or replacing the function outright:
https://github.com/qmk/qmk_firmware/blob/985b8cab7ca7103aee2b83696824683bd5159888/quantum/matrix_common.c#L86

File: keyboards/massdrop/alt/keymaps/urbanvanilla/config.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
```
Comment (zvecr) [inline keyboards/massdrop/alt/keymaps/urbanvanilla/config.h:19]:
```suggestion

#define MATRIX_IO_DELAY 40

```

**Review by zvecr — CHANGES_REQUESTED** [review]
With #11472, matrix,.c can be removed and the suggestion applied to get the same behaviour.

File: keyboards/massdrop/alt/keymaps/urbanvanilla/matrix.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (urbanvanilla) [reply keyboards/massdrop/alt/keymaps/urbanvanilla/matrix.c]: sorted it out with the later review  by @zvecr, thanks!

**Comment by drashna** [comment]
This has a bunch of existing commits, that is messing up the diff and such.  
It looks like this may need a rebase to clean stuff up.

File: keyboards/massdrop/alt/keymaps/urbanvanilla/keymap.c [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        _______, _______, RGB_TOG, _______, _______, MD_BOOT, NK_TOGG, DBG_TOG, _______, _______, _______, _______,          KC_PGUP, KC_VOLD, \
+        _______, _______, _______,                            _______,                            _______, _______, KC_HOME, KC_PGDN, KC_END  \
+    ),
+    /*
+    [X] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, \
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, \
+        _______, _______, _______,                            _______,                            _______, _______, _______, _______, _______  \
```
Comment (fauxpark) [inline keyboards/massdrop/alt/keymaps/urbanvanilla/keymap.c:34]:
```suggestion
        KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_HOME,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,  KC_PGUP,
        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_UP,   KC_PGDN,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, MO(1),   KC_LEFT, KC_DOWN, KC_RGHT
    ),
    [1] = LAYOUT_65_ansi_blocker(
        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  VLK_TOG, KC_MUTE,
        _______, RGB_SPD, RGB_VAI, RGB_SPI, RGB_HUI, RGB_SAI, _______, U_T_AUTO,U_T_AGCR,_______, KC_PSCR, KC_SLCK, KC_PAUS, _______, KC_END,
        _______, RGB_RMOD,RGB_VAD, RGB_MOD, RGB_HUD, RGB_SAD, _______, _______, _______, _______, _______, _______,          _______, KC_VOLU,
        _______, _______, RGB_TOG, _______, _______, MD_BOOT, NK_TOGG, DBG_TOG, _______, _______, _______, _______,          KC_PGUP, KC_VOLD,
        _______, _______, _______,                            _______,                            _______, _______, KC_HOME, KC_PGDN, KC_END
    ),
    /*
    [X] = LAYOUT(
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
        _______, _______, _______,                            _______,                            _______, _______, _______, _______, _______
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Comment by urbanvanilla** [comment]
Thanks!

**Pull request merged**